pyuvsim/uvsim.py

```diff
@@ -348,6 +349,37 @@ def serial_gather(uvtask_list, uv_out):
     return uv_out
 
 
+def _check_ntasks_valid(Ntasks_tot):
+    """Check that the size of the task array won't overflow the gather."""
+
+    class _spoofed_skymodel:
+        spectral_type = 'flat'
+    Ntasks_test = 35
+    tasks = [UVTask(_spoofed_skymodel(), None, None, None, None, None)
+             for _ in range(Ntasks_test)]
+
+    # Spoof values. This is all that's left on the UVTask when gathering.
+    for task in tasks:
+        task.visibility_vector = np.ones(4).astype(complex)
+        task.uvdata_index = (10, 0, 50)
+        del task.time
+        del task.freq
+        del task.freq_i
+        del task.sources
+        del task.baseline
+        del task.telescope
+
+    # MPI can handle an array of at most 2**31 bits
+    INT_MAX_BYTES = 2**31 // 8
+
+    # Need to know the change in total serialized list size from adding a new task.
+    delta_size = (len(pickle.dumps(tasks)) - len(pickle.dumps(tasks[:-1])))
+
+    size = (Ntasks_tot + 1) * delta_size
+    if size >= INT_MAX_BYTES:
+        raise ValueError(f"Too many tasks for MPI to gather successfully: {Ntasks_tot}")
```

Comment:
Can we make this a bit more helpful? Maybe suggest a fix (splitting it and combining with UVData after the fact) and point to the issue about making multiple gathers.

Reply:
Is this any clearer?

Reply:
yes, thanks!